From c51282f2f0a2f7b96aaf4b0446541801e470ce66 Mon Sep 17 00:00:00 2001
From: Vaibhav Gumashta <vgumashta@apache.org>
Date: Tue, 3 Mar 2015 18:38:56 +0000
Subject: [PATCH 0066/1164] HIVE-9831: HiveServer2 should use
 ConcurrentHashMap in ThreadFactory (Vaibhav
 Gumashta reviewed by Thejas Nair)

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1663762 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit ea810b38b3ef65fbd831c41722e2ca6e51801b54)
---
 .../server/ThreadFactoryWithGarbageCleanup.java    |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/service/src/java/org/apache/hive/service/server/ThreadFactoryWithGarbageCleanup.java b/service/src/java/org/apache/hive/service/server/ThreadFactoryWithGarbageCleanup.java
index ec19abc..fb8141a 100644
--- a/service/src/java/org/apache/hive/service/server/ThreadFactoryWithGarbageCleanup.java
+++ b/service/src/java/org/apache/hive/service/server/ThreadFactoryWithGarbageCleanup.java
@@ -19,8 +19,8 @@
 
 package org.apache.hive.service.server;
 
-import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ThreadFactory;
 
 import org.apache.hadoop.hive.metastore.RawStore;
@@ -43,7 +43,7 @@
  */
 public class ThreadFactoryWithGarbageCleanup implements ThreadFactory {
 
-  private static Map<Long, RawStore> threadRawStoreMap = new HashMap<Long, RawStore>();
+  private static Map<Long, RawStore> threadRawStoreMap = new ConcurrentHashMap<Long, RawStore>();
 
   private final String namePrefix;
 
-- 
1.7.9.5

